File: VermogensKompass/VermogensKompass/Models/DisplayCurrency.swift
import Foundation

enum DisplayCurrency: String, CaseIterable, Identifiable, Codable {
    case eur = "EUR"
    case usd = "USD"

    var id: String { rawValue }
    var code: String { rawValue }

    var title: String {
        switch self {
        case .eur: return "Euro (€)"
        case .usd: return "US-Dollar ($)"
        }
    }

    init?(currencyCode: String) {
        self.init(rawValue: currencyCode.uppercased())
    }
}

File: android/core-model/src/main/java/de/vibecode/crisis/core/model/DisplayCurrency.kt
package de.vibecode.crisis.core.model

import kotlinx.serialization.Serializable

@Serializable
enum class DisplayCurrency(val code: String, val title: String) {
    EUR("EUR", "Euro (€)"),
    USD("USD", "US-Dollar ($)");

    companion object {
        fun fromCode(code: String): DisplayCurrency? {
            return entries.firstOrNull { it.code.equals(code, ignoreCase = true) }
        }
    }
}
